docs/realtime.md

```diff
@@ -0,0 +1,145 @@
+# Realtime docs
+
+## Context
+
+### Definitions
+
+- **Events feed:** the feed of occurring events. Most of them will come from couchdb, some jobs might also trigger them. There is two types of feeds:
+  - **continuous** allows to follow events as they occurs
+  - **interval** allow the see the history of the feed from any given time
+
+
+- **Realtime:** user experienced updates of the interface from change happening from another source. **Ie,** *I have a folder opened in cozy-files on the browser, I take some pictures from my smartphone, the pictures appears in the folder without me needing to refresh the browser tab.*
+
+### What couchdb offers
+
+Couchdb supports with its `_changes` API both events feeds types:
+- using `since=now&continuous=true` we get all events **continuous**ly as they happen (SSE)
+- using `since=(last known seq_number)` we get all changes in the **interval** between last known `seq_number` and now.
+
+Couchdb also offers a `_db_updates` route, which give us **continuous** changes at the database level. This routes does not support a since parameter, as there is no global `seq_number`.
+
+### Performance limitation
+
+**We cannot have a continuous _changes feed open to every databases**
+
+## Use cases for inverval events feeds
+
+### Replication
+
+Couchdb replication algorithm can work in one-shot mode, where it replicates changes since last sync up until now, or in continuous mode where it replicates changes as they happens.
+
+- The stack will not allow continuous mode for replication.
+- This is already supported with the `_changes` route
+
+### Sharing
+
+Our sharing is based on couchdb replication rules, so also depends on `_changes` feed to ensure all changes have been applied.
+
+## Use cases for continuous events feeds
+
+### Realtime
+
+Some events should be send to the client to update views as data change.
+
+### `@event` jobs trigger
+
+Some event will trigger the activation of a job (ie. When a photo has been uploaded, generate thumbnails and extract EXIF metadatas). This should be done as soon as possible after the events
+
+### Sharing ?
+
+While not absolutely necessary, having cozy A notify cozy B when a shared document is changed allows for both better user experience (faster propagation) and better performance (no need to poll every X minutes, the N cozy we are sharing from).
+
+### Logging ?
```

Comment:
The idea is that we will probably have a log.Debug everywhere we have a realtime.Event, but it's not really relevant. Removing it.